### PR TITLE
Make analyze once test not depend on test order or flutter create command

### PR DESCRIPTION
**Pull request by dnfield**
Makes the analyze_once tests able to run in any order, removing dependencies on state set by prior tests.

Removes the test around creation, and instead just creates the necessary files for analysis.

Necessary for #47963

### CODE REVIEW

**Comment by dnfield** [comment]
This passes locally :(

**Comment by dnfield** [comment]
Ahh I have it reproducing offline.  Something is out of sync.  will fix.

**Comment by dnfield** [comment]
CI is passing now - this is ready for review.

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart [hunk 1]
```diff
@@ -109,7 +118,7 @@ void main() {
         toolExit: true,
       );
```
Comment (zanderso) [inline packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: Do you need to revert the change to libMain? Probably under a `finally {}`

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart [hunk 2]
```diff
@@ -138,8 +157,9 @@ void main() {
         exitMessageContains: '3 issues found.',
         toolExit: true,
       );
+      optionsFile.deleteSync();
```
Comment (zanderso) [inline packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: This should probably go under a `finally {}`.

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart [hunk 3]
```diff
@@ -176,7 +196,7 @@ void bar() {
         tryToDelete(tempDir);
```
Comment (zanderso) [inline packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: Maybe use a different variable name to distinguish from the `tempDir` in the outer scope.

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart [hunk 4]
```diff
@@ -196,7 +216,7 @@ StringBuffer bar = StringBuffer('baz');
...
```
Comment (zanderso) [inline packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: ditto

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart [hunk 5]
```diff
@@ -217,7 +237,7 @@ StringBuffer bar = StringBuffer('baz');
...
```
Comment (zanderso) [inline packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: ditto

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: It gets re-created in `setUp` now

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart (hunk 2 above)
Comment (dnfield) [reply packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: Done

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart (hunk 3 above)
Comment (dnfield) [reply packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: I re-grouped things so there's no more shadowing/access to unused variables.

Comment (dnfield) [reply packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: I re-shuffled things so that there is no more shadowing of this variable (but didn't rename it).

I've also made it so we don't need to call pub anymore in this test.

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart [hunk 6]
```diff
@@ -23,128 +24,162 @@ final Map<Type, Generator> noColorTerminalOverride = <Type, Generator>{
 
 void main() {
   final String analyzerSeparator = platform.isWindows ? '-' : '•';
+  const Pub pub = Pub();
```
Comment (zanderso) [inline packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: Is this pub going to do anything? Can it be a mock that we verify has the right calls made against it? If it doesn't do anything that's being tested here, should the analyze command take the --no-pub flag?

Comment (dnfield) [reply packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart]: Added --no-pub flag and removed references to this

**Review by zanderso — APPROVED** [review]
